### ISSUE TITLE
Flattened 1-table snowball output

### ISSUE THREAD

**Issue by yjunechoe** [massimoaria/openalexR#23]
_Moving this out of #9 as a separate issue_

## Input (from output of `oa_snowball()`)

Let's say we conduct a snowball search of papers B and D which finds A and C, where A cites B and C is both cited by B and cites D.

### Search graph

```mermaid
graph LR;
  B -. forward .->A
  B -- backward --> C
  D -. forward .-> C
  
style A fill:#fff
style C fill:#fff
```

### Edges

| from | to |
|------|----|
| A    | B  |
| B    | C  |
| C    | D  |

### Nodes

| id | ... | oa_input |
|----|-----|----------|
| A  | ... | FALSE    |
| B  | ... | TRUE     |
| C  | ... | FALSE    |
| D  | ... | TRUE     |

## Proposed output (of `to_disk()`)

The motivation behind `to_disk()` is to augment the paper metadata in `$nodes` with additional information about connectivity/relevance/importance of each paper in `$edges`, in a single-table format for interactive use.

What if `to_disk()` adds a column called `connection` which is semicolon-separated values of input ids that are discovered it. This collapses cites/cited_by relationships and just shows info about "proximity" to the input set. So for the above graph the `to_disk()` representation would look something like:

| id | ... | oa_input | connections |
|----|-----|----------|-------------|
| A  | ... | FALSE    | B           |
| B  | ... | TRUE     |             |
| C  | ... | FALSE    | B; D        |
| D  | ... | TRUE     |             |

If inputs cite each other, `connections` is also a place where that information can be stored. Additionally we might also add a `n_connections` column for ease of sorting when researchers are prioritizing papers for screening in excel/googlesheets.

**Comment by trangdata** [comment]
Thank you so much @yjunechoe for writing this out. I like this idea a lot, but I'd like to propose a few more columns to preserve a little more information:

| id | ... | cited_by_count | ... | oa_input | cited_by | citing | citing_count | biblicon | biblicon_count |
|----|-----|----------------|-----|----------|----------|--------|--------------|----------|----------------|
| A  | ... | 0              | ... | FALSE    |          | B      | 1            | B        | 1              |
| B  | ... | 1              | ... | TRUE     | A        | C      | 1            | A;C      | 2              |
| C  | ... | 1              | ... | FALSE    | B        | D      | 1            | B;D      | 2              |
| D  | ... | 1              | ... | TRUE     | C        |        | 0            | C        | 1              |

A few notes:
- `cited_by_count` is already a column in the nodes dataframe, so I wanted to follow this `*_count` convention
- The last three columns can be easily computed from the other columns. Is it useful to still generate them for the user?
- I'm not familiar with bibliometrics studies, @massimoaria indicated that a single ";" (no space after) would be the preferred separator
- `connection` is a bit too generic, so I wanted to find a name that is specific for citation relationships. Is `biblicon` an okay name?

**Comment by yjunechoe** [comment]
Works for me re: adding more columns than just `connections`!

One "big idea" for `to_disk()` that at least I have in mind is that it should kinda be like `show_works()`: it's doesn't have to be so principled; it should just have stuff that (non-programmer) researchers can interactively work with in a spreadsheet. What I've found is that redundant information are worth "spelling out" if they they are useful for sorting or filtering. So like adding `n_connections` (or `biblicon_count` if you prefer) was a request I got even when `connections` was already a column. So I think extra columns are fine and people can just `select()` what they want before actually writing it to a file (or delete columns in spreadsheet after-the-fact) - but open to more discussion on this

Agree with all of your other points!

**Comment by trangdata** [comment]
Thanks for explaining this, @yjunechoe! It makes more sense to me now what the purpose of this output is. I have re-implemented to_disk with [detailed documention](https://github.com/massimoaria/openalexR/blob/bd68668990dc3d0619a05e1684d57e28d4b8f6ed/R/oa_snowball.R#L127-L139) but let me know if you think we can explain something better!

**Comment by trangdata** [comment]
Resolved in #24
